.github/scripts/check-release.sh

```diff
@@ -1,24 +1,31 @@
-#!/bin/bash
+#!/usr/bin/env bash
+set -eu -o pipefail
 
-# check_tag $current_tag $file_tag $file_name
-function check_tag {
-  if [[ "$1" != "$2" ]]; then
-      echo "Error: the current tag does not match the version in Cargo.toml: found $2 - expected $1"
-      ret=1
-  fi
+check_tag() {
+    local expected=$1
+    local actual=$2
+    local filename=$3
+
+    if [[ $actual != $expected ]]; then
+        echo >&2 "Error: the current tag does not match the version in $filename: found $actual, expected $expected"
+        return 1
+    fi
+}
+
+read_version() {
+    grep '^version = ' | cut -d \" -f 2
 }
 
 ret=0
 current_tag=${GITHUB_REF#'refs/tags/v'}
 
-file_tag="$(grep '^version = ' Cargo.toml | cut -d '=' -f 2 | tr -d '"' | tr -d ' ')"
-check_tag $current_tag $file_tag
+file_tag="$(cat Cargo.toml | read_version)"
+check_tag $current_tag $file_tag Cargo.toml || ret=1
```

Comment:
Explicit is better than implicit.[©](https://peps.python.org/pep-0020/) That's why we set `ret` here instead of letting it be side-affected by `check_tag`.